tests/test_asgi.py

```diff
@@ -47,6 +53,14 @@ def test_get_js_static_file(application, test_files):
     assert send.headers[b"content-length"] == str(len(test_files.js_content)).encode()
 
 
+def test_redirect_preserves_query_string(application, test_files):
+    scope = AsgiScopeEmulator({"path": "/static/with-index", "query_string": b"v=1"})
+    receive = AsgiReceiveEmulator()
+    send = AsgiSendEmulator()
+    asyncio.run(application(scope, receive, send))
+    assert send.headers[b"location"] == b"with-index/?v=1"
```

Comment:
Make this test slightly more robust by adding an second parameter into the query string, such as `with-index/?v=1&x=2`